server/py/services/api/api/endpoints/operations.py

```diff
@@ -76,6 +82,42 @@ async def trigger_migrations(
     return background_task
 
 
+@router.post(
+    "/refresh_smtp_configuration",
+    responses={
+        http.HTTPStatus.OK.value: {},
+        http.HTTPStatus.ACCEPTED.value: {"model": mlrun.common.schemas.BackgroundTask},
+    },
+)
+async def refresh_smtp_configuration(
+    background_tasks: fastapi.BackgroundTasks,
+    response: fastapi.Response,
+    request: fastapi.Request,
+    auth_info: mlrun.common.schemas.AuthInfo = Depends(
+        framework.api.deps.authenticate_request
+    ),
+):
+    if not framework.utils.singletons.k8s.get_k8s_helper().running_inside_kubernetes_cluster:
+        raise mlrun.errors.MLRunPreconditionFailedError(
+            "SMTP configuration can be refreshed only when running inside a Kubernetes cluster"
+        )
+
+    task_callback, background_task, task_name = await run_in_threadpool(
+        _create_refresh_smtp_configuration_background_task,
+        auth_info.session,
+    )
+
+    if not background_task:
+        # No task in progress, creating a new one
+        background_tasks.add_task(task_callback)
+        background_task = framework.utils.background_tasks.InternalBackgroundTasksHandler().get_background_task(
+            task_name
+        )
+
+    response.status_code = http.HTTPStatus.ACCEPTED.value
+    return background_task
```

Comment:
There are 2 major issues here
1. if you send this request to multiple workers/chief at the same time you might hit 409 when saving the k8s secret
2. in addition to that, when you refresh from secret to worker/chiefs, it will happen only on the service that handles the requests and not all mlrun api services. 

What I think for (2) is we can add a periodic (Configurable) cache invalidation where upon each default of 30s we would read the secret